### ISSUE TITLE
How's the maintenance outlook for this library?

### ISSUE THREAD

**Issue by efokschaner** [metricsgraphics/metrics-graphics#943]
As gathered from the discussion in https://github.com/metricsgraphics/metrics-graphics/pull/929 this library seems to be going through an ownership transfer.

I'm wondering how that's going? How's the progress towards an official 3.0 release? Does @jens-ox foresee themselves being able to steward this project at least until another steward steps up? Are fixes / contributions from the community able to merged + released?

Or more to the point, would the current stewards recommend new projects use this library?

I don't mean to exert pressure, and I commend everyone that's contributed to the project so far. It just would be helpful to have a clear understanding of the likely state of stewardship, so people can decide whether it makes sense to use this lib.

**Comment by jens-ox** [comment]
Hi! Yeah, working on MG has been on my ToDo list for a while. The codebase is in quite a good shape, but the docs still need a lot of work. There are also some smaller bugs here and there.

Personally I'm not currently using MG for any projects. I did the rewrite during the first lockdown, as I was on leave and wanted to try out a bigger rewrite for quite some time.

The landscape of visualization libraries is currently quite crowded, and even though most libraries target specific frontend stacks (e.g. [visx](https://airbnb.io/visx/) for React), there are still libraries like [Observable Plot](https://github.com/observablehq/plot), which pretty much fully overlaps MG.

Long story short - I'm currently not fully sure if putting effort into MG is worth it given the alternatives like Observable Plot. If you think I'm wrong there, please let me know. Then I will build a roadmap for MG this weekend ☺️

**Comment by efokschaner** [comment]
Thanks @jens-ox for the prompt and candid reply. 

I personally don't have an informed opinion as to whether this project still satisfies a good niche in this space, so I would not urge you to take on any roadmap work.

Perhaps users with more experience using this and the other libraries you mention might weigh in?

In the meantime I was also considering Observable Plot, and now adding visx to my exploration. If you have even more suggestions I'd love to hear them.

Have a great weekend whatever you end up doing!

**Comment by formido** [comment]
Personally I think there is a niche for MG. Observable Plot is a lot, whereas this project focuses on just the most common charts, which is usually what you need. Not coincidentally, the chart on the homepage of the marketing site was precisely what I wanted for my project. And maybe because it's focused, the details are slightly nicer. I like the polish and aesthetics of the examples in this project better than Observable Plot. The function parameters were obvious and brain-dead simple. I had zero complaints about the assumptions made when chart was rendered.

Anyway, my 2¢ dropping in off of Google...

**Comment by formido** [comment]
Nine months ago, Observable Plot was introduced on Hacker News and this comment from the thread I think captures what I love about MG and the niche it fills:

![image](https://user-images.githubusercontent.com/22295/152859486-9007dfa3-2791-46de-a4d0-8135360c6c45.png)

Only, JS *doesn't* lack this, MG does it perfectly!

Even the Mozilla Metrics team's blog post explaining the design philosophy behind MG I think still really resonates. There's a reason this repo has so many stars. I use these charts for internal reporting and I don't want to have to figure out channels and scales and such when I just need time series or bar charts drawn in the usual way that my users expect.

**Comment by jens-ox** [comment]
Thanks for the kind words @formido. It took some time due to job changes etc on my side, but I'm back ☺️ 
Here's my plans for the next weeks:
* I think I still don't have push rights for NPM. @wlach I wrote you a mail some time ago regarding push rights to NPM and the domain. I will publish a beta v3 release as soon as I'm able to.
* Switch the docs over to Next.js, so I can nicely publish MDX docs.
* Add API docs.
* Add tests.
* Add the usual CI/CD stuff for testing and publishing packages on release.

**Comment by wlach** [comment]
> I think I still don't have push rights for NPM. @wlach I wrote you a mail some time ago regarding push rights to NPM and the domain. I will publish a beta v3 release as soon as I'm able to.

Hi @jens-ox, I just checked and you should have rights to push the package (jens-ox is listed as a maintainer on GitHub).

For the domain, can you contact me offline? I think we should be able to figure something out. Send me an email at `wlach@protonmail.com` or (preferably) ping me on Element (`@wlach:matrix.org`).
